### PR TITLE
dvovk/limit mem usage

### PR DESCRIPTION
**Pull request by dvovk**
Implemented limit for saving peers in an Erigon node memory to be able to turn on diagnostics data collection by default.

### CODE REVIEW

File: erigon-lib/diagnostics/network.go [hunk 1]
```diff
@@ -2,60 +2,99 @@ package diagnostics
...
+	"sync"
+	"time"
 
 	"github.com/ledgerwatch/log/v3"
 )
 
-func (d *DiagnosticClient) setupNetworkDiagnostics(rootCtx context.Context) {
-	d.runCollectPeersStatistics(rootCtx)
+type PeerStats struct {
+	peersInfo     sync.Map
```
Comment (AskAlexSharov) [inline erigon-lib/diagnostics/network.go:13]: i think must be *sync.Map

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

File: erigon-lib/diagnostics/network.go (hunk 1 above)
Comment (dvovk) [reply erigon-lib/diagnostics/network.go:13]: Why ? I don't think that will make any difference.

Comment (AskAlexSharov) [reply erigon-lib/diagnostics/network.go:13]: I don't know how to right use sync.Map - I mean https://go.dev/tour/methods/8

Comment (dvovk) [reply erigon-lib/diagnostics/network.go]: Done

**Pull request merged**